pkg/virtual/generic/apiserver/apiserver.go

```diff
@@ -0,0 +1,107 @@
+package apiserver
```

Comment:
Why `pkg/virtual/generic/apiserver/`? Will there be a non-generic virtual? Will there be something other than an APIserver in here?

Reply:
There is:
- on 1 side only one *generic* package that contains the abstracted logic that allows building virtual workspace API-servers (rooted at some URL prefix) with the minimal amount of code.
- On the other side several packages for concrete virtual workspace implementations. More precisely one package per concrete virtual workspace implementation rooted at a given prefix. For now there is only one implementation of a concrete virtual workspace for the `workspaces` resources. But there would be others, for `APIExportView` resources for example.

So the whole idea of the generic part is to implement once for all common parts and logic of virtual workspaces, and make the concrete implementation of concrete virtual workspaces easy (mainly dedicated initialization and REST storage implementation), without having to cope with the APIServer initialization and delegation anymore.

Reply:
cc @sttts

Reply:
we might drop the "virtual", wdyt?

Reply:
`virtual` for everything related to *virtual workspaces*